src/transformStateWithClones.js

```diff
@@ -7,7 +7,61 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+
+  for (const action of actions) {
+    const lastItemInArray = result[result.length - 1];
+
+    switch (action.type) {
+      case 'addProperties':
+        let addedProperties;
+
+        if (result.length === 0) {
+          addedProperties = Object.assign({}, state, action.extraData);
+        } else {
+          addedProperties
+            = Object.assign({}, lastItemInArray, action.extraData);
+        }
+
+        result.push(addedProperties);
```

Comment:
I suggest to short it a little bit.
If there is no elements in array your variable will be equal  `undefined` so you don't need to check result.length, just check is there a currentStateObject
```suggestion
        result.push(Object.assign({}, currentStateObject || state, action.extraData));
```